sequencing-server/src/builders/seqJsonBuilder.ts

```diff
@@ -0,0 +1,113 @@
+import { ActivateStep, CommandStem, LoadStep, Sequence } from '../lib/codegen/CommandEDSLPreface.js';
+import type { SeqBuilder } from '../types/seqBuilder.js';
+
+export type Command = CommandStem | ActivateStep | LoadStep;
+
+export type SeqJsonBuilder = SeqBuilder<Command[], Sequence>;
+
+export const seqJsonBuilder: SeqJsonBuilder = (expandedActivities, seqId, seqMetadata, simulationDatasetId) => {
+  let allCommands: (CommandStem | ActivateStep | LoadStep)[] = [];
+  let convertedCommands: (CommandStem | ActivateStep | LoadStep)[] = [];
+  let activityInstanceCount = 0;
+  let planId;
+  // Keep track if we should try and sort the commands.
+  let shouldSort = true;
+  let timeSorted = false;
+  let previousTime: Temporal.Instant | undefined = undefined;
+
+  for (const ai of expandedActivities) {
+    // If errors, no associated Expansion
+    if (ai.errors !== null) {
+      planId = ai?.simulationDataset.simulation?.planId;
+
+      if (ai.errors.length > 0) {
+        allCommands = allCommands.concat(
+          ai.errors.map(e =>
+            CommandStem.new({
+              stem: '$$ERROR$$',
+              arguments: [{ message: e.message }],
+            }).METADATA({ simulatedActivityId: ai.id }),
+          ),
+        );
+      }
+
+      // Typeguard only
+      if (ai.expansionResult === null) {
+        break;
+      }
+
+      /**
+       * Treat the activity start time as the "previous time" for sorting expanded commands that begin with a command-relative time tag.
+       *
+       * TODO: we argue that this is actually ideal behavior, but it warrants discussion.
+       */
+      previousTime = ai.startTime;
+      for (const command of ai.expansionResult) {
+        const currentCommand =
+          command instanceof CommandStem
+            ? (command as CommandStem)
+            : command instanceof LoadStep
+            ? (command as LoadStep)
+            : (command as ActivateStep);
```

Comment:
I'm still fairly new to TS, but would this work?
```suggestion
        const currentCommand = command as (CommandStem | LoadStep | ActivateStep);
```

Reply:
Fixed!